src/C/common.jl

```diff
@@ -636,24 +412,22 @@ const OGRGeometryH = Ptr{Void}
 const OGRSpatialReferenceH = Ptr{Void}
 const OGRCoordinateTransformationH = Ptr{Void}
 
-type _CPLXMLNode
+mutable struct _CPLXMLNode
 end
 
 const OGRFieldDefnH = Ptr{Void}
 const OGRFeatureDefnH = Ptr{Void}
 const OGRFeatureH = Ptr{Void}
 const OGRStyleTableH = Ptr{Void}
 
-type OGRGeomFieldDefnHS
+mutable struct OGRGeomFieldDefnH
```

Comment:
how are these meant to be used?

```julia
julia> GDAL.getnameref(GDAL.OGRGeomFieldDefnH())
ERROR: ccall: the type of argument 1 doesn't correspond to a C type
Stacktrace:
 [1] getnameref(::GDAL.OGRGeomFieldDefnH) at /Users/yeesian/.julia/v0.6/GDAL/src/ogr_api.jl:2506

julia> GDAL.gettype(GDAL.OGRGeomFieldDefnH())
ERROR: ccall: the type of argument 1 doesn't correspond to a C type
Stacktrace:
 [1] gettype(::GDAL.OGRGeomFieldDefnH) at /Users/yeesian/.julia/v0.6/GDAL/src/ogr_api.jl:2522
```

Reply:
Good point, don't know. I assumed that it would work since Clang.jl wraps them like this now, but perhaps I should manually force it back in `types.jl`, the old way.

I believe it is wrapped differently since in C it is `typedef struct OGRGeomFieldDefnHS *` as opposed to just `typedef void *`.

Reply:
It was something that surfaced in ArchGDAL as I was updating it. I think there's a `OGRGeomFieldDefnHS` and `OGRGeomFieldDefnH` is a pointer to it.

Reply:
Could you check my edits in 62b243dbde35916de718738d6392b8fee89744b1?
Otherwise I believe this PR is ready.

Reply:
I can confirm that all the tests are passing locally on https://github.com/yeesian/ArchGDAL.jl/pull/20 based on the changes here now

Reply:
I should mention I'm very happy with the consistency in the pointer types now, and with the change to use enums

Reply:
I'm very happy with you pointing out any inconsistencies :)